content/help/ops/traffic-management/deploy-guidelines/index.md

```diff
@@ -6,6 +6,113 @@ weight: 5
+host in a single `VirtualService` or `DestinationRule` resource, it may be possible to incrementally specify
+the configuration for the host in separate resources, which will be merged when they are applied.
+
+Consider the case of a `VirtualService` bound to an ingress gateway exposing an application host which uses
+path-based delegation to several implementation services, something like this:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: myapp
+spec:
+  hosts:
+  - myapp.com
+  gateways:
+  - myapp-gateway
+  http:
+  - match:
+    - uri:
+        prefix: /service1
+    route:
+    - destination:
+        host: service1.default.svc.cluster.local
+  - match:
+    - uri:
+        prefix: /service2
+    route:
+    - destination:
+        host: service2.default.svc.cluster.local
+  - match:
+    ...
+{{< /text >}}
+
+The downside of this kind of configuration is that other configuration (e.g., route rules) for any of the
+underlying microservices, will need to also be included in this single configuration file, instead of
+in separate resources associated with, and potentially owned by, the individual service teams.
+See [Route rules have no effect on ingress gateway requests](#route-rules-have-no-effect-on-ingress-gateway-requests)
+for details.
+
+To avoid this problem, it may be preferable to break up the configuration of `myapp.com` into several
+`VirtualService` fragments, one per implementation service. For example:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: myapp-service1
+spec:
+  hosts:
+  - myapp.com
+  gateways:
+  - myapp-gateway
+  http:
+  - match:
+    - uri:
+        prefix: /service1
+    route:
+    - destination:
+        host: service1.default.svc.cluster.local
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: myapp-service2
+spec:
+  hosts:
+  - myapp.com
+  gateways:
+  - myapp-gateway
+  http:
+  - match:
+    - uri:
+        prefix: /service2
+    route:
+    - destination:
+        host: service2.default.svc.cluster.local
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: myapp-...
+{{< /text >}}
+
+When a second and subsequent `VirtualService` for an existing host is applied, istio-pilot will merge
+the additional route rules into the existing configuration of the host. There are, however, several
+limitations with this feature that must be considered carefully when using it.
+
+1. Although the order of evaluation for rules in any given source `VirtualService` will be retained,
+   the cross-resource order is UNDEFINED. In other words, there is no guaranteed order of evaluation
+   for rules across the fragment configurations, so it will only have predictable behavior if there
+   are no conflicting rules or order dependency between rules across fragments.
```

Comment:
may be add an example of a conflicting rule?

Reply:
I think there are too many possible examples to choose any one that is particularly instructive.